### PR TITLE
Data Service Deployment Tests

### PR DESCRIPTION
**Pull request by Madanagopal19**
**What this PR does / why we need it**: This PR has the PDS testcase to deploy data services
No of scenarios: 4

1. Deployment OnDemand, Deploy desired Image of a Data Service Version
Ex ENV Param
export DEPLOY_ALL_DATASERVICE=false
export DEPLOY_ALL_VERSIONS=false
export DATA_SERVICE=Redis
export DS_BUILD="27396ca"
export DS_VERSION="7.0.4"

2. Deployment OnDemand, Deploy All Supported Versions and supported Images of a Data Service
Ex ENV Param
export DEPLOY_ALL_DATASERVICE=false
export DEPLOY_ALL_VERSIONS=true
export DATA_SERVICE=Redis

3. Deploy all data services with all available Versions and Images
Ex ENV Param
export DEPLOY_ALL_DATASERVICE=true

4. Deploy all Images of data service version
Ex ENV Param
export DEPLOY_ALL_DATASERVICE=false
export DEPLOY_ALL_VERSIONS=false
export DEPLOY_ALL_IMAGES=true
export DATA_SERVICE=Redis
export DS_VERSION="7.0.4"

Testcase Execution logs:
https://gist.github.com/Madanagopal19/afc68fd8c96844946fa639a198434c73

### CODE REVIEW

File: tests/basic/sharedv4_test.go [hunk 1]
```diff
@@ -6,12 +6,12 @@ import (
 	"strings"
 	"time"
 
-	"github.com/libopenstorage/openstorage/pkg/mount"
 	"github.com/portworx/sched-ops/k8s/apps"
 	"github.com/portworx/torpedo/drivers/node"
 	"github.com/portworx/torpedo/drivers/scheduler"
 	"github.com/portworx/torpedo/pkg/testrailuttils"
 	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/sig-storage-lib-external-provisioner/v6/mount"
```
Comment (lsrinivas-pure) [inline tests/basic/sharedv4_test.go:14]: make you are not changing the packages by mistake

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: drivers/pds/api/dataServiceDeployment.go [hunk 2]
```diff
@@ -112,20 +112,20 @@ func (ds *DataServiceDeployment) GetDeployment(deploymentID string) (*pds.Models
 }
 
 // GetDeploymentSatus return deployment status.
-func (ds *DataServiceDeployment) GetDeploymentSatus(deploymentID string) (*pds.ControllersStatusResponse, error) {
+func (ds *DataServiceDeployment) GetDeploymentSatus(deploymentID string) (*pds.ControllersStatusResponse, *status.Response, error) {
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply drivers/pds/api/dataServiceDeployment.go]: done

File: tests/pds/dataservice_test.go [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+			Step("Get the resource and app config template for supported dataservice", func() {
+				dataServiceDefaultResourceTemplateIDMap, dataServiceNameIDMap, err = pdslib.GetResourceTemplate(tenantID, supportedDataServices)
+				Expect(err).NotTo(HaveOccurred())
+
+				dataServiceNameDefaultAppConfigMap, err = pdslib.GetAppConfTemplate(tenantID, dataServiceNameIDMap)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(dataServiceNameDefaultAppConfigMap).NotTo(BeEmpty())
+			})
+		} else {
+			Expect(DeployAllDataService).To(Equal(true))
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply tests/pds/dataservice_test.go]: done, I have removed this redundant validation.

File: tests/basic/sharedv4_test.go (hunk 1 above)
Comment (Madanagopal19) [reply tests/basic/sharedv4_test.go:14]: yeah, Its my bad. I have added the commented lines back.

**Review by aravindranath14 — APPROVED** [review]
LGTM

**Comment by Madanagopal19** [comment]
Regression run Passed - https://jenkins.pwx.dev.purestorage.com/job/Dev/job/torpedo-dev01/176/parameters/

**Pull request merged**